UIKitMovie/UIKitMovie/ViewModel/ListMovie/ListMovieViewModel.swift

```diff
@@ -0,0 +1,71 @@
+// ListMovieViewModel.swift
+// Copyright © RoadMap. All rights reserved.
+
+import Foundation
+
+/// Вью модель списка фильмов
+final class ListMovieViewModel: ListMovieViewModelProtocol {
+    // MARK: - Public Properties
+
+    var movies: [Movie] = []
+    var updateView: (() -> ())?
+    var errorAlert: ((Error) -> ())?
+    var listMovieStates: ((ListMovieStates) -> ())?
+    var networkService: NetworkServiceProtocol
+    var imageService: ImageServiceProtocol
+
+    // MARK: - Initializers
+
+    init(
+        networkService: NetworkServiceProtocol,
+        imageService: ImageServiceProtocol
+    ) {
+        self.networkService = networkService
+        self.imageService = imageService
+    }
+
+    // MARK: - Public Methods
+
+    func segmentControlAction(index: Int) {
+        switch index {
+        case 0:
+            fetchMovies(method: .upcomingMethod)
+        case 1:
+            fetchMovies(method: .popularMethod)
+        case 2:
+            fetchMovies(method: .topRatedMethod)
+        default:
+            break
+        }
+    }
+
+    func fetchMovies() {
+        fetchMovies(method: .upcomingMethod)
+    }
+
+    func fetchImage(url: String, completion: @escaping (Result<Data, Error>) -> Void) {
+        imageService.getImage(url: url) { result in
+            switch result {
+            case let .success(data):
+                completion(.success(data))
```

Comment:
```suggestion
                completion(.success(data))
```
вместо этого кложуры должны быть